numpy/core/src/umath/ufunc_object.c

```diff
@@ -3970,17 +3971,26 @@ PyUFunc_GenericReduction(PyUFuncObject *ufunc, PyObject *args,
         }
     }
     else {
-        if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|OO&O&i:reduce",
+        if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|OO&Oi:reduce",
                                         reduce_kwlist,
                                         &op,
                                         &axes_in,
                                         PyArray_DescrConverter2, &otype,
-                                        PyArray_OutputConverter, &out,
+                                        &out_obj,
                                         &keepdims)) {
             Py_XDECREF(otype);
             return NULL;
         }
     }
+    /* if out_obj is a tuple with size 1 (=nout here), unpack it */
+    if (out_obj != NULL && PyTuple_CheckExact(out_obj) &&
+            PyTuple_GET_SIZE(out_obj) == 1) {
+        out_obj = PyTuple_GET_ITEM(out_obj, 0);
+    }
```

Comment:
Would be nice to give an explicit error here if the output is a tuple of the wrong length - elsewhere, we have:
```
if (PyTuple_GET_SIZE(value) != nout) {
    PyErr_SetString(PyExc_ValueError,
            "The 'out' tuple must have exactly "
            "one entry per ufunc output");
    goto fail;
}
```